compiler/rustc_typeck/src/check/fn_ctxt/mod.rs

```diff
@@ -26,6 +26,8 @@ use rustc_trait_selection::traits::{ObligationCause, ObligationCauseCode};
 use std::cell::{Cell, RefCell};
 use std::ops::Deref;
 
+/// The `FnCtxt` stores type-checking context needed to type-check function bodies,
+/// in contrast to [`ItemCtxt`], which is used to type-check item *signatures*.
```

Comment:
It's a bit more accurate to mention:
1. that this is used to lower things that don't have function bodies. For example, `ItemCtxt` is used to check structs and stuff too.
2. This is also (via the `AstConv` trait) primarily used for _lowering_ things from HIR -> middle representations of things. For example, we use it in the `tcx.type_of` query, which does little checking and mostly lowering.

Reply:
Thanks! I added the info you suggested.

The one thing is that I couldn't find an `InferCtxt` field -- or any mention of `InferCtxt` for that matter -- in `FnCtxt`. Is it accessed in some indirect way?

Reply:
> Is it accessed in some indirect way

Yea, good question. It's stored in [`inh: &Inherited`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_typeck/check/struct.FnCtxt.html#structfield.inh) -- this is how several `FnCtxt`s actually share the same `InferCtxt`. This `Inherited` struct is important because closures can share inference variables and other info to their containing functions, for example.

Reply:
Ah, that makes sense! Thanks :)